### DISCUSSION TITLE
Exception with line number

### DISCUSSION THREAD

**Discussion by jimtng** [boc-tothefuture/openhab-jruby#129] [Ideas]
I did a quick test to compare jython vs jruby while `log:set INFO org.openhab.core.automation` i.e. not set to DEBUG

When I have an erroneous line in jython, I get this in the log
```
21:49:01.163 [ERROR] [ript.internal.ScriptEngineManagerImpl] - Error during evaluation of script 'file:/openhab/conf/automation/jsr223/python/personal/test.py': NameError: name 'asdf' is not defined in <script> at line number 32
```

In jruby:
```
21:49:19.368 [ERROR] [ript.internal.ScriptEngineManagerImpl] - Error during evaluation of script 'file:/openhab/conf/automation/jsr223/ruby/personal/test.rb': org.jruby.embed.EvalFailedException: (NameError) undefined local variable or method `asfd' for main:Object
```

Is there a way to include the line number in that error message? I know this may be a jruby thing, not within the scope of our library?

**Comment by boc-tothefuture** [comment]
It is [here](https://github.com/jruby/jruby/wiki/RedBridge#Line_Number) but it looks like we should get line numbers

**Comment by boc-tothefuture** [comment]
I think this is duplicate of the open issue:  https://github.com/boc-tothefuture/openhab-jruby/issues/17

See that issue for latest update from JRuby devs.

**Comment by boc-tothefuture** [comment]
@jim and @pacive you may want to weigh in here:. https://github.com/jruby/jruby/issues/6560
